### PR TITLE
refactor(auth-api): Rename SupportsDelegation to SupportsCustomDelegation

### PR DESCRIPTION
**Pull request by saevarma**
https://www.notion.so/eb83399457da4c41a2afeabfb178dcb9?v=d180c0b047504d689094757376bba9db&p=a106ff052b8e472cb3beb3e084048d6e&pm=s

## What

Renaming the `SupportsDelegation` to `SupportsCustomDelegation`

## Why

For clarity as the `SupportsDelegation` can be misunderstood as general support for delegations, when it is only meaning the custom delegations.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/auth-api-lib/migrations/20220823071133-rename-supportsdelegation-to-supportscustomdelegation.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+'use strict'
+
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    queryInterface.renameColumn(
+      'client',
+      'supports_delegation',
+      'supports_custom_delegation',
+    )
```
Comment (eirikurn) [inline libs/auth-api-lib/migrations/20220823071133-rename-supportsdelegation-to-supportscustomdelegation.js:9]: This migration could cause a lot of pain during release. Can we create a new column instead with the same data as supports_delegation, and clean up the old column later?

**Review by eirikurn — CHANGES_REQUESTED** [review]

File: libs/auth-api-lib/migrations/20220823071133-rename-supportsdelegation-to-supportscustomdelegation.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      defaultValue: false,
+    })
+
+    await queryInterface.sequelize.query(`
+      UPDATE client SET supports_custom_delegation = supports_delegation
+    `)
+  },
+
+  async down(queryInterface, Sequelize) {
+    queryInterface.removeColumn('client', 'supports_custom_delegation')
```
Comment (eirikurn) [inline libs/auth-api-lib/migrations/20220823071133-rename-supportsdelegation-to-supportscustomdelegation.js:17]: Should this be awaited?

```suggestion
    await queryInterface.removeColumn('client', 'supports_custom_delegation')
```

Comment (saevarma) [reply libs/auth-api-lib/migrations/20220823071133-rename-supportsdelegation-to-supportscustomdelegation.js]: Absolutely

**Review by eirikurn — APPROVED** [review]
Sweet!

**Pull request merged**